src/app/providers/datatables.provider.js

```diff
@@ -63,48 +63,87 @@ class DataTablesProvider {
 
         // Display all fields other than excluded fields
         Object.keys(data).sort().forEach((key) => {
+            let formattedKey = key.charAt(0).toUpperCase() + key.substr(1).toLowerCase();
+            let value = '';
+            if (key === 'SOURCE') {
+                value = `<a href="${data[key]}" target="_blank">${data[key]}</a>`;
+            } else {
+                value = data[key];
+            }
             if (excludedFields.indexOf(key) < 0) {
-                let formatedKey = key.charAt(0).toUpperCase() + key.substr(1).toLowerCase();
-
-                extraInfo += `<li><b>${formatedKey}</b><ul><li>${data[key]}</li></ul></li>`;
+                extraInfoItems.push(`<li><b>${formattedKey}</b><ul><li>${value}</li></ul></li>`);
+            } else {
+                if (index.primary[0] && (key === index.primary[0]) && value) {
+                    firstExtraInfo = `<li><b>${formattedKey}</b><ul><li>${value}</li></ul></li>`;
+                }
+                if (index.primary[1] && (key === index.primary[1]) && value) {
+                    secondExtraInfo = `<li><b>${formattedKey}</b><ul><li>${value}</li></ul></li>`;
+                }
             }
             // --As a part of order online button--
-            if (key == '_ID') {
-                let formatedKeyId = key.replace(/\w\S*/g, (txt) => {
+            if (key === '_ID') {
+                let formattedKeyId = key.replace(/\w\S*/g, (txt) => {
                     return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
                 });
-                urlLink += `&${formatedKeyId}=${data[key]}`;
+                urlLink += `&${formattedKeyId}=${data[key]}`;
             }
-            // -- End OnlineButn--
+            // -- End OnlineButton--
         });
+        extraInfo += firstExtraInfo + secondExtraInfo + extraInfoItems.join('');
         // --As a part of order online button--
         urlLink += `&resource_id=${data['RESOURCE ID']}`;
 
+        attribute1 = data['GIVEN NAME/S'];
+        attribute2 = data['ITEM ID'];
+        attribute3 = data['SOURCE'];
+
         if (index.itemTitleField.trim() !== '')
             urlLink += `&itemTitle=${encodeURIComponent(index.itemTitleField)}`;
 
         // Display an Order Online button
-        extraInfo += `</ul><form class="form form-button">
-                            <a href=${urlLink}><button class="qsa-button">Order Online</button></a>
+        extraInfo += `</ul><form class="form form-button order-form" action="https://test.smartservice.qld.gov.au/services/prodi/addProduct" method="post">
+      <fieldset>
+        Quantity: <input type="text" name="quantity" value="1" id="quantity" size="2" ng-change="vm.changeOrderType()" ng-model="vm.quantity" />
+      </fieldset>
+      <fieldset ng-show="vm.price !== ''">
+        Price: <span ng-bind="vm.price"></span>
+      </fieldset>
+
+      <input type="hidden" name="attribute1" value="${attribute1}" />
```

Comment:
I think using a **Directive** would be much better in this case?
Directives are **easy to debug and test**

Reply:
I amended the code in the way already happening, but are you suggesting us to refactor this method so that we don't see HTML Tags?

Reply:
Yes I think with Angular it is good if we have directives to encapsulate functions (HTML, js , css)
This helps to debug easily , add new functionality and writing test cases.

But if that is difficult to separate then we can leave as it is.